store/splitstore.go

```diff
@@ -0,0 +1,93 @@
+package store
+
+import (
+	"github.com/loomnetwork/go-loom/plugin"
+)
+
+type splitStore struct {
+	KVReader
+	VersionedKVStore
+	deleted map[string]bool
+	version int64
+}
+
+func NewSplitStore(full KVReader, empty VersionedKVStore, version int64) VersionedKVStore {
+	return &splitStore{
+		KVReader:         full,
+		VersionedKVStore: empty,
```

Comment:
Seeing as this mutable `VersionedKVStore` is supposed to be empty you should just create it here instead of passing it in.